Classification/FeatureSelection/src/main/java/org/tribuo/classification/fs/FS_Wrapper_Approaches/Optimizers/CuckooSearchOptimizer.java

```diff
@@ -0,0 +1,157 @@
+        return setOfSolutions;
+    }
+
+    /**
+     * Does this feature selection algorithm return an ordered feature set?
+     *
+     * @return True if the set is ordered.
+     */
+    @Override
+    public boolean isOrdered() {
+        return true;
+    }
+
+    /**
+     * Selects features according to this selection algorithm from the specified dataset.
+     * @param dataset The dataset to use.
+     * @return A selected feature set.
+     */
+    @Override
+    public SelectedFeatureSet select(Dataset<Label> dataset) {
+        ImmutableFeatureMap FMap = new ImmutableFeatureMap(dataset.getFeatureMap());
+        setOfSolutions = GeneratePopulation(dataset.getFeatureMap().size());
+        List<FeatureSet_FScore_Container> subSet_fScores = new ArrayList<>();
+        SelectedFeatureSet selectedFeatureSet = null;
+        // Update the solution based on the levy flight function
+        for (int i = 0; i < maxIteration; i++) {
+            IntStream.range(0, setOfSolutions.length).parallel().forEach(subSet -> {
+                AtomicInteger currentIter = new AtomicInteger(subSet);
+                int[] evolvedSolution = Arrays.stream(setOfSolutions[subSet]).map(x -> Binarizing.discreteValue(transferFunction, x + stepSizeScaling * Math.pow(currentIter.get() + 1, -lambda))).toArray();
+                int[] randomCuckoo = setOfSolutions[new Random().nextInt(setOfSolutions.length)];
+                if (FitnessFunction.EvaluateSolution(this, dataset, FMap, evolvedSolution) > FitnessFunction.EvaluateSolution(this, dataset, FMap, randomCuckoo))
+                    System.arraycopy(evolvedSolution, 0, setOfSolutions[subSet], 0, evolvedSolution.length);
+                // Update the solution based on the abandone nest function
+                if (new Random().nextDouble() < worstNestProbability) {
+                    int r1 = new Random().nextInt(setOfSolutions.length);
+                    int r2 = new Random().nextInt(setOfSolutions.length);
+                    for (var j = 0; j < setOfSolutions[subSet].length; j++)
+                        evolvedSolution[j] = Binarizing.discreteValue(transferFunction, setOfSolutions[subSet][j] + delta * (setOfSolutions[r1][j] - setOfSolutions[r2][j]));
+                    if (FitnessFunction.EvaluateSolution(this, dataset, FMap, evolvedSolution) > FitnessFunction.EvaluateSolution(this, dataset, FMap, setOfSolutions[subSet]))
+                        System.arraycopy(evolvedSolution, 0, setOfSolutions[subSet], 0, evolvedSolution.length);
+                }
+                subSet_fScores.add(new FeatureSet_FScore_Container(setOfSolutions[subSet], FitnessFunction.EvaluateSolution(this, dataset, FMap, setOfSolutions[subSet])));
+            });
+            subSet_fScores.sort(Comparator.comparing(FeatureSet_FScore_Container::score).reversed());
+            selectedFeatureSet = FitnessFunction.getSFS(this, dataset, FMap, subSet_fScores.get(0).subSet);

```

Comment:
Does this need to create the SFS each iteration? It looks like this could be called once after the loop finishes.